src/main/java/uk/gov/hmcts/reform/sscs/transform/deserialize/SubmitYourAppealToCcdCaseDataDeserializer.java

```diff
@@ -99,9 +102,32 @@ public static SscsCaseData convertSyaToCcdCaseData(SyaCaseWrapper syaCaseWrapper
             .benefitCode(benefitCode)
             .issueCode(issueCode)
             .caseCode(caseCode)
+            .dwpRegionalCentre(getDwpRegionalCenterGivenDwpIssuingOffice(appeal.getBenefitType().getCode(),
+                appeal.getMrnDetails().getDwpIssuingOffice()))
             .build();
     }
 
+    private static String getDwpRegionalCenterGivenDwpIssuingOffice(String benefitTypeCode, String dwpIssuingOffice) {
+        if ("UC".equals(benefitTypeCode)) {
+            DwpAddressLookupService dwpAddressLookupService = new DwpAddressLookupService();
+            Optional<OfficeMapping> officeMapping = dwpAddressLookupService.getDwpMappingByOffice(
+                benefitTypeCode, dwpIssuingOffice);
+            return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
+        }
+        if (dwpIssuingOffice != null && benefitTypeCode != null) {
+            DwpAddressLookupService dwpAddressLookupService = new DwpAddressLookupService();
+            Optional<OfficeMapping> officeMapping = dwpAddressLookupService.getDwpMappingByOffice(
+                benefitTypeCode, dwpIssuingOffice);
+            if ("ESA".equals(benefitTypeCode)) {
+                return officeMapping.map(mapping -> mapping.getMapping().getCcd()).orElse(null);
```

Comment:
well, it's actually repeated twice because for the PIP it is slightly different and we return the mapping.dwpRegionCenter when for UC and ESA we return mapping.ccd. 
I could refactor the ESA and UC to use a method but then I will repeat this method twice.

Reply:
ok